server/bin/pbench-server-put-shim-001.py

```diff
@@ -0,0 +1,176 @@
+# submits for further processing. It copies the tarballs and their MD5
+# sums to the archive (after checking) and sets the state links, so
+# that the dispatch script will pick them up and get the ball
+# rolling.
+
+import os
+import sys
+import glob
+import tempfile
+
+from logging import Logger
+from pathlib import Path
+from collections import namedtuple
+
+from pbench.agent.results import CopyResultTb
+from pbench.common.exceptions import BadConfig
+from pbench.common.logger import get_pbench_logger
+from pbench.common.utils import md5sum
+from pbench.server import PbenchServerConfig
+from pbench.server.report import Report
+
+
+_NAME_ = "pbench-server-prep-shim-002"
+
+Results = namedtuple("Results", ["nstatus", "ntotal", "ntbs"])
+
+
+def get_receive_dir(config: PbenchServerConfig, logger: Logger) -> Path:
+
+    # we are explicitly handling version-002 data in this shim
+    receive_dir_prefix = config.get("pbench-server", "pbench-receive-dir-prefix")
+    if not receive_dir_prefix:
+        logger.error("Failed: config value pbench-receive-dir-prefix pbench-server")
+        return None
+
+    receive_dir = Path(f"{receive_dir_prefix}-002").resolve()
+    if not receive_dir.is_dir():
+        logger.error("Failed: {} does not exist, or is not a directory", receive_dir)
+        return None
+
+    return receive_dir
+
+
+def process_tb(
+    config: PbenchServerConfig, logger: Logger, receive_dir: Path
+) -> Results:
+    """process_tb - receives a tar ball and copies it to the remote server
+                    using the server's HTTP PUT method
+
+        Args -
+            config -- PbenchServer config object
+            logger -- logger object to use when emitting log entries during
+                      operation
+            receive_dir -- directory where tarballs are received from agent
+
+    """
+
+    # Check for results that are ready for processing: version 002 agents
+    # upload the MD5 file as xxx.md5.check and they rename it to xxx.md5
+    # after they are done with MD5 checking so that's what we look for.
+    list_check = glob.glob(
+        os.path.join(receive_dir, "**", "*.tar.xz.md5"), recursive=True
+    )
+
+    logger.info("{}", config.TS)
+    list_check.sort()
+    nstatus = ""
+
+    ntotal = ntbs = 0
+
+    # FIXME: keep a generated token which has longer duration
+
+    token = config.get("pbench-server", "put-token")
+
+    for tbmd5 in list_check:
+        ntotal += 1
+
+        # full pathname of tarball
+        tb = Path(tbmd5[0:-4])
+        tbmd5 = Path(tbmd5)
+        tbdir = tb.parent
+        controller = tbdir.name
+
+        try:
+            tarball_len, tarball_md5 = md5sum(tb)
+            crt = CopyResultTb(
+                controller, tb, tarball_len, tarball_md5, config, logger,
+            )
+            crt.copy_result_tb(token)
+        except Exception as e:
+            logger.error("Unexpected Error: {}", e)
```

Comment:
What is `config.TS` and why isn't it logged here, as it is in lines 102 and 111?

Reply:
config.TS gives the live timestamp when that particular statement is run. I will add that to this logger as well